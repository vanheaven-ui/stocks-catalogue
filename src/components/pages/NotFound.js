// import { useHistory } from 'react-router-dom';

import { Link } from 'react-router-dom';

const NotFound = () => (
  <section id="404" className="not-found" style={{ color: '#fff' }}>
    404 Page is not found
    <Link to="/"><span role="img" aria-labelledby="404">🏠 Go Home</span></Link>
  </section>
);
export default NotFound;
