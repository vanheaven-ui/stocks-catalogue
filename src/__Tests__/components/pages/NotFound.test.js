import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import NotFound from '../../../components/pages/NotFound';
import rootNode from '../../helpers';

describe('NotFound page', () => {
  test('Renders without crashing', () => {
    ReactDOM.render(<BrowserRouter><NotFound /></BrowserRouter>, rootNode());
  });

  test('doesnot change unexpectedly', () => {
    const notfoundTree = renderer.create(<BrowserRouter><NotFound /></BrowserRouter>).toJSON();
    expect(notfoundTree).toMatchSnapshot();
  });

  test('renders the correct wrapper element', () => {
    const { container } = render(<BrowserRouter><NotFound /></BrowserRouter>);
    expect(container.firstChild.classList.contains('not-found')).toBe(true);
  });
});
