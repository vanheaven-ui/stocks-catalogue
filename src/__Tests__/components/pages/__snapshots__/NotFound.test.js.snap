// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound page doesnot change unexpectedly 1`] = `
<section
  className="not-found"
  id="404"
  style={
    Object {
      "color": "#fff",
    }
  }
>
  404 Page is not found
  <a
    href="/"
    onClick={[Function]}
  >
    <span
      aria-labelledby="404"
      role="img"
    >
      🏠 Go Home
    </span>
  </a>
</section>
`;
